ReportPortal.Tests/ReportPortal.E2E.API.Business/StepDefinitions/ApiControllers/ProjectController.cs

```diff
@@ -15,39 +14,39 @@ public Task<HttpResponseMessage> CreateProject(string projectName, string entryT
             };
             var endpoint = Endpoints.ManageProject;
             Log.LogInformation($"CreateProject '{projectName}' with EntryType '{entryType}'\n Method: Post\n Endpoint: {endpoint}");
-            return _launchApiSteps.PostAsJsonAsync(endpoint, requestBody);
+            return _launchApiSteps.Post<T>(endpoint, requestBody);
         }
 
         public void DeleteProject(int id)
         {
             var endpoint = $"{Endpoints.ManageProject}/{id}";
             Log.LogInformation($"DeleteProject with Id '{id}'\n Method: Delete\n Endpoint: {endpoint}");
-            _launchApiSteps.DeleteAsync(endpoint);
+            _launchApiSteps.Delete(endpoint);
         }
 
-        public Task<HttpResponseMessage> GetProjectsList()
+        public T GetProjectsList<T>()
         {
             var endpoint = $"{Endpoints.GetProjectList}";
             Log.LogInformation($"GetProjectsList \n Method: Get\n Endpoint: {endpoint}");
-            return _launchApiSteps.GetAsync(endpoint);
+            return _launchApiSteps.Get<T>(endpoint);
         }
 
-        public Task<HttpResponseMessage> SearchProjectUser(string projectName, string userName)
+        public T SearchProjectUser<T>(string projectName, string userName)
         {
             var endpoint = string.Format(Endpoints.SearchProjectUser, projectName, userName);
             Log.LogInformation($"SearchProjectUser \n Method: Get\n Endpoint: {endpoint}");
-            return _launchApiSteps.GetAsync(endpoint);
+            return _launchApiSteps.Get<T>(endpoint);
         }
       
-        public Task<HttpResponseMessage> AddUserToProject(string projectName, string userName, string userRole = "MEMBER")
+        public T AddUserToProject<T>(string projectName, string userName, string userRole = "MEMBER")
         {
```

Comment:
May be for userRole enum will be better?

Reply:
Added enum